### PR TITLE
Duplo 3732, DUPLO-3752 and DUPLO-3731.

### PR DESCRIPTION
**Pull request by tahir-duplo**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Pull requests may not be submitted to the *master* branch (use *develop* instead) - or they will be closed.
- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer.
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-3731 Add azure data source to get storage account keys.](https://app.clickup.com/t/8655600/DUPLO-3731)

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-3732 Add azure resource for storage_share_file.](https://app.clickup.com/t/8655600/DUPLO-3732)

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-3731 Add azure data source to get storage account keys.](https://app.clickup.com/t/8655600/DUPLO-3731)

File: duplocloud/resource_duplo_azure_storage_share_file.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	c := m.(*duplosdk.Client)
+	duplo, clientErr := c.StorageShareFileGet(tenantID, storageAccountName, name)
+	if clientErr != nil {
+		if clientErr.Status() == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Unable to retrieve tenant %s azure storage share file %s : %s", tenantID, name, clientErr)
+	}
+
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_azure_storage_share_file.go:80]: check for `nil` - duplo does not always return a 404

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_azure_storage_share_file.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_azure_storage_share_file.go:84]: Done.

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-3752 TF: check for nil against all v1 duplo read APIs.](https://app.clickup.com/t/8655600/DUPLO-3752)

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**